File: 45-jump-game-ii/45-jump-game-ii.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        int n = nums.size();
        
        int ans = 0;
        int left = 0, right = 0;
        int farthest = 0;
        
        for(int i=0; i<n-1; i++){
            farthest = max(farthest, i+nums[i]);
            if(i==right){
                ans++;
                right = farthest;
            }
        }
        
        return ans;
    }
};

File: 45-jump-game-ii/45-jump-game-ii.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        
        n, r, farthest, ans = len(nums), 0, 0, 0
        
        for i in range(n-1):
            farthest = max(farthest, i + nums[i])
            
            if i == r:
                ans += 1
                r = farthest
        return ans